tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TagMap.java

```diff
@@ -41,55 +45,49 @@
  * Note that we expect fairly small tag maps which are iterated over, not used for lookups. Most {@link Map}
  * methods are implemented, but use a naive linear search rather than a binary search.
  */
+@Immutable
 @SuppressWarnings("JdkObsolete")
 final class TagMap implements SortedMap<String, String> {
+    static final TagMap EMPTY = TagMap.of(ImmutableMap.of());
 
-    static final TagMap EMPTY = new TagMap(new String[0]);
-
-    private final String[] values;
-    private final int hash;
+    private final ImmutableList<String> values;
```

Comment:
It may be wroth taking a look at this change with [JOL](https://openjdk.java.net/projects/code-tools/jol/) to compare the difference for an average size tag map.